File: C/917. Reverse Only Letters.c
char * reverseOnlyLetters(char * s){
    char* l = s;
    char* r = s + strlen(s) - 1;
    char temp;
    while(l <= r){
        if(!isalpha(*l)){
            l++;
        }
        else if(!isalpha(*r)){
            r--;
        }
        else{
            temp = *l;
            *l = *r;
            *r = temp;
            l++;
            r--;
        }
    }
    return s;
}

File: JavaScript/917. Reverse Only Letters.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseOnlyLetters = function(s) {
    var string = s.split("");
    var l = 0;
    var r = string.length - 1;
    while(l <= r){
        if(!(/[a-zA-Z]/.test(string[l]))){
            l++;
        }
        else if(!(/[a-zA-Z]/.test(string[r]))){
            r--;
        }
        else{
            var temp = string[l];
            string[l++] = string[r];
            string[r--] = temp;
        }
    }
    return string.join('');
};
